easybuild/tools/filetools.py

```diff
@@ -792,6 +792,57 @@ def find_easyconfigs(path, ignore_dirs=None):
     return files
 
 
+def locate_files(files, paths, ignore_subdirs=None):
+    """
+    Determine full path for list of files, in given list of paths (directories).
+    """
+    # determine which easyconfigs files need to be found, if any
+    files_to_find = []
+    for idx, ec_file in enumerate(files):
+        if ec_file == os.path.basename(ec_file) and not os.path.exists(ec_file):
+            files_to_find.append((idx, ec_file))
+    _log.debug("List of files to find: %s" % files_to_find)
+
+    # find missing easyconfigs by walking paths in robot search path
+    for path in paths:
+        _log.debug("Looking for missing files (%d left) in %s..." % (len(files_to_find), path))
+
+        # try to load index for current path, or create one
+        path_index = load_index(path, ignore_dirs=ignore_subdirs)
+        if path_index is None or build_option('ignore_index'):
+            if os.path.exists(path):
+                _log.info("No index found for %s, creating one...", path)
+                path_index = create_index(path, ignore_dirs=ignore_subdirs)
```

Comment:
Not sure what create_index does exactly, but if it is creating an index file, does it verify that it actually can (permissions) and if it doesn't then what?
For instance, our local ec repo tree doesn't have an index and is not writable by users, so what happens if a user runs this?

And the log message above, should not say "creating one" if it isn't allowed to...

Reply:
`create_index` just creates an index in memory, it doesn't dump it to file (that's done via `dump_index`), so as long as there's read access, that's enough

Reply:
ok